test/integration/targets/vmware_host_dns/tasks/static.yml

```diff
@@ -0,0 +1,212 @@
+    search_domains: "{{ dns['results'][0]['hosts_dns_info'][esxi1]['search_domain'] }}"
+
+# Testcase 0006: Ensure DNS config on the cluster is idempotent for static
+- name: Ensure static DNS config is idempotent when done on the cluster
+  vmware_host_dns:
+    hostname: '{{ vcenter_hostname }}'
+    username: '{{ vcenter_username }}'
+    password: '{{ vcenter_password }}'
+    cluster_name: "{{ ccr1 }}"
+    validate_certs: False
+    type: 'static'
+    domain: "{{ dns['results'][0]['hosts_dns_info'][esxi1]['domain_name'] }}"
+    dns_servers: "{{ dns['results'][0]['hosts_dns_info'][esxi1]['ip_address'] }}"
+    search_domains: "{{ dns['results'][0]['hosts_dns_info'][esxi1]['search_domain'] }}"
+  register: vmware_host_dns_result_0006
+
+- name: Ensure DNS config wasn't changed
+  assert:
+    that:
+      - vmware_host_dns_result_0006 is not changed
+
+# Testcase 0007: Ensure changing the domain on the cluster works
+- name: Ensure changing the domain on the cluster works
+  vmware_host_dns:
+    hostname: '{{ vcenter_hostname }}'
+    username: '{{ vcenter_username }}'
+    password: '{{ vcenter_password }}'
+    cluster_name: "{{ ccr1 }}"
+    validate_certs: False
+    type: 'static'
+    domain: new.domain
+    dns_servers: "{{ dns['results'][0]['hosts_dns_info'][esxi1]['ip_address'] }}"
+    search_domains: "{{ dns['results'][0]['hosts_dns_info'][esxi1]['search_domain'] }}"
+  register: vmware_host_dns_result_0007
+
+- name: Ensure DNS config was changed
+  assert:
+    that:
+      - vmware_host_dns_result_0007 is changed
+
+# Testcase 0008: Ensure changing the DNS servers on the cluster works
+- name: Ensure changing the DNS servers on the cluster works
+  vmware_host_dns:
+    hostname: '{{ vcenter_hostname }}'
+    username: '{{ vcenter_username }}'
+    password: '{{ vcenter_password }}'
+    cluster_name: "{{ ccr1 }}"
+    validate_certs: False
+    type: 'static'
+    domain: new.domain
+    dns_servers:
+      - 1.2.3.4
+      - 5.6.7.8
+    search_domains: "{{ dns['results'][0]['hosts_dns_info'][esxi1]['search_domain'] }}"
+  register: vmware_host_dns_result_0008
+
+- name: Ensure DNS config was changed
+  assert:
+    that:
+      - vmware_host_dns_result_0008 is changed
+
+# Testcase 0009: Ensure changing the search domains on the cluster works
+- name: Ensure changing the search domains on the cluster works
+  vmware_host_dns:
+    hostname: '{{ vcenter_hostname }}'
+    username: '{{ vcenter_username }}'
+    password: '{{ vcenter_password }}'
+    cluster_name: "{{ ccr1 }}"
+    validate_certs: False
+    type: 'static'
+    domain: new.domain
+    dns_servers:
+      - 1.2.3.4
+      - 5.6.7.8
+    search_domains:
+      - subdomain.example.local
+      - example.local
+  register: vmware_host_dns_result_0009
+
+- name: Ensure DNS config was changed
+  assert:
+    that:
+      - vmware_host_dns_result_0009 is changed
+
+# Revert to original DNS configuration
+- name: Revert to original DNS configuration
```

Comment:
This should be in an `always` section of a `block`. This way, if the test fails, the correct name will be put back.

Reply:
I've moved the tests into a `block` and the task "Revert to original DNS configuration" into an `always` section.

I've also added a changelog fragment and the rest of the documentation to deprecate `vmware_dns_config`.